api/src/schemas.ts

```diff
@@ -196,5 +193,47 @@ export const schemas = {
         })
       })
     }
+  },
+  // Challenges:
+  projectCompleted: {
+    body: Type.Object({
+      id: Type.String({ format: 'objectid', maxLength: 24 }),
+      challengeType: Type.Optional(Type.Number()),
+      solution: Type.String({ format: 'url', maxLength: 1024 }),
+      // TODO(Post-MVP): require format: 'url' for githubLink
+      githubLink: Type.Optional(Type.String())
+    }),
+    response: {
+      200: Type.Object({
+        // TODO(Post-MVP): delete completedDate and alreadyCompleted? As far as
+        // I can tell, they are not used anywhere
+        completedDate: Type.Number(),
+        points: Type.Number(),
+        alreadyCompleted: Type.Boolean()
+      }),
+      400: Type.Object({
+        type: Type.Literal('error'),
+        message: Type.Union([
+          Type.Literal(
+            'That does not appear to be a valid challenge submission.'
+          ),
+          Type.Literal(
+            'You have not provided the valid links for us to inspect your work.'
+          )
+        ])
+      }),
+      403: Type.Object({
+        type: Type.Literal('error'),
+        message: Type.Literal(
+          'You have to complete the project before you can submit a URL.'
+        )
+      }),
```

Comment:
Shouldn't they all be 403 errors?

Reply:
Nope. In the first case they're being rejected because the request is missing something we need to process it. That means 400 is more appropriate.

In the second case, yeah, 403 makes sense because the server understands the request; it refuses to do what it's being asked as the user hasn't satisfied the prerequisites.

Reply:
Gotcha. My doubt came up because they were all 403 errors in the old API.

Reply:
Yeah, I had the same doubts for the same reasons.

Fortunately the client doesn't care about the codes, so we can fix them without updating the client.